src/mmap.rs

```diff
@@ -525,6 +560,34 @@ impl<B: NewBitmap> GuestMemoryMmap<B> {
     }
 }
 
+#[cfg(feature = "xen")]
+impl<B: NewBitmap> GuestMemoryMmap<B> {
+    /// Creates a container and allocates anonymous memory for guest memory regions.
+    ///
+    /// Valid memory regions are specified as a sequence of (Address, Size, Option<FileOffset>)
+    /// tuples sorted by Address.
+    pub fn from_ranges_with_files<A, T>(ranges: T) -> result::Result<Self, Error>
+    where
+        A: Borrow<(GuestAddress, usize, Option<FileOffset>, u32, u32)>,
+        T: IntoIterator<Item = A>,
+    {
+        Self::from_regions(
+            ranges
+                .into_iter()
+                .map(|x| {
+                    GuestRegionMmap::from_range(
+                        x.borrow().0,
+                        x.borrow().1,
+                        x.borrow().2.clone(),
+                        x.borrow().3,
+                        x.borrow().4,
```

Comment:
Would it maybe make sense to introduce a struct instead of a tuple? I guess rust-analyzer's code lenses help when reading it, but I feel this is still becoming a fairly long list of ints with various meanings.